package.json

```diff
@@ -42,6 +42,7 @@
   },
   "dependencies": {
     "jszip": "^2.3.0",
+    "when": "^3.7.0",
```

Comment:
We should limit the number of dependencies, so people are not forced to include when.js (which is of considerable size). Consider the situation when they already have e.g. bluebird. We may consider switching to callback approach or using smaller library/polyfill, such as https://github.com/calvinmetcalf/lie